test/distributed/test_functional_api.py

```diff
@@ -31,10 +31,22 @@
     instantiate_parametrized_tests,
     parametrize,
     run_tests,
+    skipIfHpu,
+    TEST_CUDA,
+    TEST_HPU,
     TestCase,
 )
 
 
+DEVICE = "cuda"
+devices = ["cpu"]
+if TEST_HPU:
+    devices.append("hpu")
+    DEVICE = "hpu"
+elif TEST_CUDA:
+    devices.append("cuda")
+
+
```

Comment:
Do you mind adding a comment here re how to set these fields? It would help future developers extend the test if they have a new device.

Reply:
What is the big letter `DEVICE` for?

Reply:
There are two types of tests in this file. One type which runs only on accelerators and the other which must be tested on both cpu and accelerator.  

The tests which require both cpu and accelerators are accessed through the "devices" list by use of the parameterize decorator as seen in this change: https://github.com/pytorch/pytorch/pull/138216/files#diff-65c626933dcc96bd93039223954541be496473ef9575b7e064055227538edd5cR249 

As you can see the previous version had hardcoded cuda and cpu, instead by passing the "devices" list we enable cleaner device agnostic use

To address the tests which use only accelerators instead I have used the "DEVICE" variable. You can see it being used here for example: https://github.com/pytorch/pytorch/pull/138216/files#diff-65c626933dcc96bd93039223954541be496473ef9575b7e064055227538edd5cR764

It is necessary to use a second variable to ensure these tests do not run on the cpu

I will add this information in the comment on how to set these fields as suggested

Reply:
comment added in: 65e16deb07c5ca9acef323058a99d26e89633acb